src/components/GearItem.tsx

```diff
@@ -0,0 +1,99 @@
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import Button from "@material-ui/core/Button";
+import ButtonGroup from "@material-ui/core/ButtonGroup";
+import React, { FunctionComponent } from "react";
+import ErrorIcon from "@material-ui/icons/Error";
+import MenuItem from "@material-ui/core/MenuItem";
+import Select from "@material-ui/core/Select";
+import Equipment from "../resources/Equipment";
+
+interface GearItemProps {
+  item: Equipment;
+  quantity: number;
+  setFieldValue: (field: string, value: number | string | boolean) => void;
+}
+const GearItem: FunctionComponent<GearItemProps> = ({
+  item,
+  quantity,
+  setFieldValue,
+}) => {
+  // Create all possible quantities for current item
+  const selectOptions: JSX.Element[] = [];
+  for (let i = 0; i <= item.quantity; ++i) {
+    selectOptions.push(<MenuItem value={i}>{i}</MenuItem>);
+  }
+
+  // Show ErrorIcon
+  const showError = (): void => {
+    const element = document.getElementById("error");
+    if (element) {
+      element.style.display = "block";
+    }
+  };
+
+  // Change current Gear Value
+  const changeValue = (newValue: number): void => {
+    if (newValue >= 0 && newValue <= item.quantity) {
+      setFieldValue("gear[" + item.title + "]", newValue);
+    }
+    if (newValue > item.quantity) {
+      showError();
+    }
+  };
```

Comment:
This is certainly a confusing one.  I think you can just put the second `if` first, no?
Note that you're defining a procedure that can branch in very different ways!  I don't think the generic "changeValue" label is appropriate - it puts a false sense of security in one's mind about what is going on here.
```suggestion
  const setGearOrShowError = (newValue: number): void => {
      if (newValue > item.quantity || newValue < 0) {
        showError();
      }
      setFieldValue("gear[" + item.title + "]", newValue);
  };
```